akka-typed/src/main/scala/akka/typed/scaladsl/Actor.scala

```diff
@@ -193,6 +192,33 @@ object Actor {
   def onSignal[T](handler: PartialFunction[(ActorContext[T], Signal), Behavior[T]]): Behavior[T] =
     immutable[T]((_, _) ⇒ same).onSignal(handler)
 
+  /**
+   * Behavior that stops all child actors, waits for them to stop and then
+   * runs some user defined cleanup before stopping itself.
+   */
+  def gracefulStop[T](cleanup: ActorContext[T] ⇒ Unit): Behavior[T] =
```

Comment:
what is the difference between `gracefulStop` and the existing stopped?

```
  /**
   * Return this behavior from message processing to signal that this actor
   * shall terminate voluntarily. If this actor has created child actors then
   * these will be stopped as part of the shutdown procedure.
   *
   * The PostStop signal that results from stopping this actor will be passed to the
   * given `postStop` behavior. All other messages and signals will effectively be
   * ignored.
   */
  def stopped[T](postStop: Behavior[T]): Behavior[T] = Behavior.stopped(postStop)
```

Isn't the `cleanup` callback like the `postStop` parameter?

Reply:
`cleanup` is run before the actor is stopped. `postStop` runs after the actor has stopped. That is how I understand this. Please correct me if I am wrong.